### DISCUSSION TITLE
passHref: always passing the href

### DISCUSSION THREAD

**Discussion by ybrodsky** [vercel/next.js#25553] [Ideas]
### What version of Next.js are you using?

10.2.3

### What version of Node.js are you using?

14.14.0

### What browser are you using?

Chrome

### What operating system are you using?

Ubuntu

### How are you deploying your application?

Irrelevant

### Describe the Bug

When using the `Link` component, the href is always passed down to the `<a>` tag.

```
<Link href="/something">
  <a>Something</a>
<Link>

// output
<a href="/something">Something</a>
```

The documentation states that `passHref` defaults to false, so one would expect the  `<a>` not to contain href. 
Anyway, even if you specify `passHref={false}` explicitly, it still passes down the href.

If you take a look here https://github.com/vercel/next.js/blob/canary/packages/next/client/link.tsx#L294 that conditiona will always be true. Either you have `passHref={true}` or the child is a `<a>` without href itself, like the example above `<a>Something</a>`.

### Expected Behavior

Should not passHref by default. Or at least not when specifying false.

### To Reproduce

Create a link

```
<Link href="/something">
  <a>Something</a>
<Link>
```

**Comment by lxsmnsyc** [comment]
According to the documentation, `passHref`'s role is to whether or not force the href to be passed to the children, specially if the direct children is not an anchor tag. I believe the behavior is documented well as well as expected.

https://nextjs.org/docs/api-reference/next/link#if-the-child-is-a-custom-component-that-wraps-an-a-tag

**Comment by ybrodsky** [comment]
Well, besides what the initial intention was with this prop, you already have a variable. Why not enforce it correctly when I say I want it to be false.
If you have links in your site that you don't care for google to index (you know rel="nofollow" is not that trustable) having `passHref={false}` can help you to stay within nextjs route without having the href in the rendered links.

**Comment by ybrodsky** [comment]
Closing an issue without bothering to reply, nice.

**Reply by timneutkens** [reply]
It's not closed, it was converted to discussions, you're asking for a feature change which falls under features which are converted to discussions.

**Reply by ybrodsky** [reply]
My bad then.
How do I move this discussion forward so that it might be considered for implementation?

Removing the second part of [this](https://github.com/vercel/next.js/blob/canary/packages/next/client/link.tsx#L290) IF would let you decide not to pass down the href to a `<a>` tag but it would also become the default, since `passHref` defaults to false.
If you change `passHref` to be default true, you maintain the standard scenario where a `<a>` inside a `<Link>` would get the href by default, which will also be true for child components that are not `<a>` (I don't see a downside to this when you can then explicitly set the prop as false).
